File: Coding Challenge/Exam Challenge/Exam coding Challenge/President.cpp
#include "President.h"
#include "Worker.h"
#include<string>
using namespace std;
using namespace pattern;
using namespace worker;

President::President(){

}
void President::Worker(){
  cout<<"I am proud of my students and university. I am happy to bestow them their degrees and wish them luck."<<endl;
}
void President::nextWorker(Worker worker){
   w = worker;
}

President::~President(){
  
}

File: Coding Challenge/Exam Challenge/worker Java Files/President.java
package pattern;

import pattern.worker.Worker;

public class President extends Worker{
	
	@Override
	public void work() {
		System.out.println("I am proud of my students and university. "
				+ "I am happy to bestow them their degrees and wish them luck.");
	}

	@Override
	public void nextWorker(Worker worker) {
		w = worker;
	}

}
